tests/test_cli/test_generate/test_protocols.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This test module contains the tests for the `aea scaffold protocol` sub-command."""
+# import filecmp
+import json
+import os
+import shutil
+import tempfile
+import unittest.mock
+from pathlib import Path
+
+import jsonschema
+# import yaml
+from ...common.click_testing import CliRunner
+from jsonschema import Draft4Validator  # ValidationError
+
+# from aea import AEA_DIR
+import aea.cli.common
+import aea.configurations.base
+# from aea.configurations.base import DEFAULT_PROTOCOL_CONFIG_FILE
+from aea.cli import cli
+from ...conftest import CLI_LOG_OPTION, PROTOCOL_CONFIGURATION_SCHEMA, CONFIGURATION_SCHEMA_DIR, CUR_PATH
+
+
+class TestGenerateProtocol:
+    """Test that the command 'aea scaffold protocol' works correctly in correct preconditions."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set the test up."""
+        cls.runner = CliRunner()
+        cls.agent_name = "myagent"
+        cls.cwd = os.getcwd()
+        cls.t = tempfile.mkdtemp()
+        shutil.copyfile(Path(CUR_PATH, "data", "sample_specification.yaml"), Path(cls.t, "sample_specification.yaml"))
+        cls.path_to_specification = str(Path("..", "sample_specification.yaml"))
+        cls.patch = unittest.mock.patch.object(aea.cli.common.logger, 'error')
+        cls.mocked_logger_error = cls.patch.__enter__()
+
+        cls.schema = json.load(open(PROTOCOL_CONFIGURATION_SCHEMA))
+        cls.resolver = jsonschema.RefResolver("file://{}/".format(Path(CONFIGURATION_SCHEMA_DIR).absolute()), cls.schema)
+        cls.validator = Draft4Validator(cls.schema, resolver=cls.resolver)
+
+        os.chdir(cls.t)
+        result = cls.runner.invoke(cli, [*CLI_LOG_OPTION, "create", cls.agent_name], standalone_mode=False)
+        assert result.exit_code == 0
+        os.chdir(cls.agent_name)
+        # scaffold protocol
+        cls.result = cls.runner.invoke(cli, [*CLI_LOG_OPTION, "generate", "protocol", cls.path_to_specification], standalone_mode=False)
+
+    def test_exit_code_equal_to_0(self):
+        """Test that the exit code is equal to 0."""
+        assert self.result.exit_code == 0
+
+    # def test_resource_folder_contains_module_message(self):
```

Comment:
The remaining tests need to be rewritten for generate, they are copied from scaffold